modules/dockerregistry/dockerregistry_test.go

```diff
@@ -0,0 +1,291 @@
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	port, ipAddress := getRegistryPortAndAddress(t, err, container, ctx)
+
+	// Let's simply check that the registry is up and running with a GET to http://localhost:5000/v2/_catalog
+	h := http.Client{}
+	req, _ := http.NewRequest("GET", "http://"+ipAddress+":"+port.Port()+"/v2/_catalog", nil)
+	resp, err := h.Do(req)
+	require.Equal(t, resp.StatusCode, 401)
+	defer resp.Body.Close()
+	_, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+}
+
+func TestDockerRegistryWithAuthAndData(t *testing.T) {
+	t.Cleanup(func() {
+		os.Setenv("DOCKER_AUTH_CONFIG", originalDockerAuthConfig)
+	})
+	os.Unsetenv("DOCKER_AUTH_CONFIG")
+
+	// using the same credentials as in the Docker Registry
+	base64 := "dGVzdHVzZXI6dGVzdHBhc3N3b3Jk" // testuser:testpassword
+	t.Setenv("DOCKER_AUTH_CONFIG", `{
+		"auths": {
+				"localhost:5000": { "username": "testuser", "password": "testpassword", "auth": "`+base64+`" }
+		},
+		"credsStore": "desktop"
+	}`)
+	ctx := context.Background()
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	container, err := RunContainer(ctx, WithAuthentication(wd+"/../../testdata/auth"), WithData(wd+"/../../testdata/data"))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	// Let's check that we are able to start a container form image localhost:5000/redis:5.0-alpine
+	// using default username and password
+	req := testcontainers.ContainerRequest{
+		Image:           "localhost:5000/redis:5.0-alpine",
+		AlwaysPullImage: true, // make sure the authentication takes place
+		ExposedPorts:    []string{"6379/tcp"},
+		WaitingFor:      wait.ForLog("Ready to accept connections"),
+	}
+
+	redisContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.Nil(t, err)
+	terminateContainerOnEnd(t, ctx, redisContainer)
+	_, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+}
+
+func TestDockerRegistryWithAuthDataAndImage(t *testing.T) {
+	t.Cleanup(func() {
+		os.Setenv("DOCKER_AUTH_CONFIG", originalDockerAuthConfig)
+	})
+	os.Unsetenv("DOCKER_AUTH_CONFIG")
+
+	// using the same credentials as in the Docker Registry
+	base64 := "dGVzdHVzZXI6dGVzdHBhc3N3b3Jk" // testuser:testpassword
+	t.Setenv("DOCKER_AUTH_CONFIG", `{
+		"auths": {
+				"localhost:5000": { "username": "testuser", "password": "testpassword", "auth": "`+base64+`" }
+		},
+		"credsStore": "desktop"
+	}`)
+	ctx := context.Background()
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	container, err := RunContainer(ctx, WithAuthentication(wd+"/../../testdata/auth"), WithData(wd+"/../../testdata/data"), WithImage("docker.io/registry:latest"))
```

Comment:
I'd copy/duplicate the resources that exist in the core library to the module, to avoid relative paths and decouple the module from the upstream core library just a little bit more.

We could decide later if we want to test that feature in the core or in the module.